src/cascadia/TerminalSettingsEditor/Appearances.h

```diff
@@ -0,0 +1,200 @@
+﻿/*++
+Copyright (c) Microsoft Corporation
+Licensed under the MIT license.
+
+Module Name:
+- Appearances
+
+Abstract:
+- The classes defined in this module are responsible for encapsulating the appearance settings
+  of profiles and presenting them in the settings UI
+
+Author(s):
+- Pankaj Bhojwani - May 2021
+
+--*/
+
+#pragma once
+
+#include "Font.g.h"
+#include "Appearances.g.h"
+#include "AppearanceViewModel.g.h"
+#include "Utils.h"
+#include "ViewModelHelpers.h"
+#include "SettingContainer.h"
+
+// Function Description:
+// - This function presents a File Open "common dialog" and returns its selected file asynchronously.
+// Parameters:
+// - customize: A lambda that receives an IFileDialog* to customize.
+// Return value:
+// (async) path to the selected item.
+template<typename TLambda>
+static winrt::Windows::Foundation::IAsyncOperation<winrt::hstring> OpenFilePicker(HWND parentHwnd, TLambda&& customize)
+{
+    auto fileDialog{ winrt::create_instance<IFileDialog>(CLSID_FileOpenDialog) };
+    DWORD flags{};
+    THROW_IF_FAILED(fileDialog->GetOptions(&flags));
+    THROW_IF_FAILED(fileDialog->SetOptions(flags | FOS_FORCEFILESYSTEM | FOS_NOCHANGEDIR | FOS_DONTADDTORECENT)); // filesystem objects only; no recent places
+    customize(fileDialog.get());
+
+    auto hr{ fileDialog->Show(parentHwnd) };
+    if (!SUCCEEDED(hr))
+    {
+        if (hr == HRESULT_FROM_WIN32(ERROR_CANCELLED))
+        {
+            co_return winrt::hstring{};
+        }
+        THROW_HR(hr);
+    }
+
+    winrt::com_ptr<IShellItem> result;
+    THROW_IF_FAILED(fileDialog->GetResult(result.put()));
+
+    wil::unique_cotaskmem_string filePath;
+    THROW_IF_FAILED(result->GetDisplayName(SIGDN_FILESYSPATH, &filePath));
+
+    co_return winrt::hstring{ filePath.get() };
+}
+
+// Function Description:
+// - Helper that opens a file picker pre-seeded with image file types.
+static winrt::Windows::Foundation::IAsyncOperation<winrt::hstring> OpenImagePicker(HWND parentHwnd)
+{
+    static constexpr COMDLG_FILTERSPEC supportedImageFileTypes[] = {
+        { L"All Supported Bitmap Types (*.jpg, *.jpeg, *.png, *.bmp, *.gif, *.tiff, *.ico)", L"*.jpg;*.jpeg;*.png;*.bmp;*.gif;*.tiff;*.ico" },
+        { L"All Files (*.*)", L"*.*" }
+    };
+
+    static constexpr winrt::guid clientGuidImagePicker{ 0x55675F54, 0x74A1, 0x4552, { 0xA3, 0x9D, 0x94, 0xAE, 0x85, 0xD8, 0xF2, 0x7A } };
+    return OpenFilePicker(parentHwnd, [](auto&& dialog) {
+        THROW_IF_FAILED(dialog->SetClientGuid(clientGuidImagePicker));
+        try
+        {
+            auto pictureFolderShellItem{ winrt::capture<IShellItem>(&SHGetKnownFolderItem, FOLDERID_PicturesLibrary, KF_FLAG_DEFAULT, nullptr) };
```

Comment:
Usually putting functions in header files is a big strict _no_. Are these being used in two places?

The one is fine, as it's a template... but this second one cannot be in a header as it would get defined twice and duplicated etc etc etc

Reply:
> Are these being used in two places?

Yes, both `Appearances.cpp` and `Profiles.cpp` use it (Appearances for choosing background image path, Profiles for choosing starting directory)

Reply:
Moved to `Utils.h` and `Utils.cpp`